cub/util_device.cuh

```diff
@@ -348,6 +366,19 @@ CUB_RUNTIME_FUNCTION inline cudaError_t PtxVersionUncached(int& ptx_version)
 
         result = cudaFuncGetAttributes(&empty_kernel_attrs,
                                        reinterpret_cast<void*>(empty_kernel));
+        // Any failure means either no GPU is available or the provided device
+        // binary does not match the generated function code
+        if (result != cudaSuccess) {
+            int sm_version;
+            int device;
+            if (!CubDebug(cudaGetDevice(&device))) {
+                if (!CubDebug(SmVersionUncached(sm_version, device))) {
+                    printf("Incompatible GPU: you are trying to run this program on sm_%d%d, "
```

Comment:
Should we include something like stdio in this header if we are going to use printf unconditionally?

Reply:
Also, should it be `fprintf(stderr, ...)` instead?